nlmod/cache.py

```diff
@@ -832,3 +879,29 @@ def ds_contains(
         coords={k: ds.coords[k] for k in coords},
         attrs={k: ds.attrs[k] for k in attrs},
     )
+
+
+def _explicit_dataset_coordinate_comparison(ds_in, ds_cache):
+    """Perform explicit dataset coordinate comparison.
+
+    Parameters
+    ----------
+    ds_in : xr.Dataset
+        Input dataset.
+    ds_cache : xr.Dataset
+        Cached dataset.
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    AssertionError
+        If the coordinates are not equal.
+    """
+    logger.info("cache -> performing explicit dataset coordinate comparison")
+    for coord in ds_cache.coords:
+        logger.debug("cache -> comparing coordinate: %s", coord)
+        assert_equal(ds_in[coord], ds_cache[coord])
```

Comment:
I think `assert_identical` is a better check. https://docs.xarray.dev/en/v0.9.2/generated/xarray.testing.assert_identical.html